backends/views.py

```diff
@@ -24,4 +26,14 @@ def complete(request, *args, **kwargs):
         request.session[key] = backend_state
 
     # Continue with social_core pipeline
-    return social_complete(request, *args, **kwargs)
+    social_complete_rtn = social_complete(request, *args, **kwargs)
+
+    # Update redis cache if user had invalid credentials
+    if request.user.is_authenticated():
+        con = get_redis_connection("redis")
+        user_key = FIELD_USER_ID_BASE_STR.format(request.user.id)
+        if con.hexists(CACHE_KEY_FAILURE_NUMS_BY_USER, user_key):
```

Comment:
It looks like this check is not necessary. If `hdel` or `srem` are operating on a value which doesn't exist it just returns a 0: https://redis.io/commands/hdel and https://redis.io/commands/srem